contrib/imkmsg/kmsg.c

```diff
@@ -220,7 +246,13 @@ readkmsg(void)
 			continue;
 		} else {
 			/* something went wrong - error or zero length message */
-			if (i < 0 && errno != EINTR && errno != EAGAIN) {
+			if(i < 0 && (errno == EAGAIN || errno == EWOULDBLOCK)) {
+				DBGPRINTF("imkmsg: initial read done, changing to non-blocking mode\n");
```

Comment:
I think this message should actually be "changing to blocking mode" (rather than "non-blocking")?

Reply:
indeed (Y)